### ISSUE TITLE
Extending itemValue leads to weird behavior

### ISSUE THREAD

**Issue by SamMousa** [surveyjs/survey-creator#3184]
### Are you requesting a feature, reporting a bug or ask a question?
Bug / feature

I'm trying to extend the item value class in the creator only.
Imagine that I have a multiple choice question type, but for each choice I want the person creator to also specify another property.

For example say we have a list of car models including brand, car models are often classified in rough categories as well.

- Tesla Model 3 -- Mid-Size
- Tesla Model X -- Large
- Mini Cooper -- Minicompact

The goal here is to use this information during analysis, it doesn't need to go in the survey data, but the survey creator should be able to store it. 

### What is the current behavior?
The survey creator renders the configuration for the choices fine:
![image](https://user-images.githubusercontent.com/547021/174829049-12cf4d0f-295f-4ca6-a82c-09dd0148fdd8.png)
(Ignore the clipping of values for now)

But it doesn't load it from the JSON.

### What is the expected behavior?
Extending the `itemvalue` class should either work or explicitly be forbidden.

### How would you reproduce the current behavior (if this is a bug)?
https://plnkr.co/edit/rsYMrS9rUDhoSnlI

#### Provide the test code and the tested page URL (if applicable)

Tested page URL: https://plnkr.co/edit/rsYMrS9rUDhoSnlI

Test code

```
// New question type for facility type:
  const widgetName = 'facilitytype'
  const facilityQuestionType = {
    name: widgetName,
    title: 'Facility Type',
    iconName: 'icon-radiogroup',
    category: 'HeRAMS',
    isFit (question) {
      return question.getType() === widgetName
    },
    isDefaultRender: true,

    widgetIsLoaded () {
      return true
    },

    init () {
      Survey.Serializer.addClass('facilityitemvalue', [
        {
          name: 'tier',
          displayName: 'Tier',
          choices: ['primary', 'secondary', 'tertiary'],
          default: 'primary'
        }, {
          name: 'text'
        }
      ], null, 'itemvalue')

      Survey.Serializer.addClass(widgetName, [
        {
          name: 'choices',
          type: 'facilityitemvalue[]'

        },
        {
          name: 'choicesFromQuestion',
          visible: false
        },
        {
          name: 'hasOther',
          visible: false
        },
        {
          name: 'hasNone',
          visible: false
        }
      ], null, 'radiogroup')
    },
    afterRender (question, el) {

    }

  }

  Survey.CustomWidgetCollection.Instance.add(facilityQuestionType, 'customtype')
```

### Specify your

* browser: Brave latest
* editor version: 1.9.36

**Comment by andrewtelnov** [comment]
Here is [the working example](https://plnkr.co/edit/O1fcLDCjcQhT4V8P).

Thank you,
Andrew

**Comment by SamMousa** [comment]
> SurveyJS assumes that properties added into addClass directly, are a common properties for a common object.

I don't understand what this means? The property in question is a common property for a common object.. Regardless, if it's not supported to be called in this way, would it not make sense to just throw an error in such cases?

**Comment by andrewtelnov** [comment]
@SamMousa I will take a look. If we create an instance of the class that doesn't really exists, we may assume that all it's properties are custom and doesn't exist in the code.

Thank you,
Andrew

**Comment by andrewtelnov** [comment]
@SamMousa Unfortunately I can't fix this issue. 
I will ask @RomanTsukanov  to update the documentation.
There is `Serializer.addClass` and `Serializer.addProperty`/`Serializer.addProperties` functions.
addClass setup the serializer for a Survey object. In most cases, there is a class in the code, like Base (base), Question (question), Page(), QuestionTextModel (text). We add ability to create a class on the fly (without defining prototype in code). 
I was thinking to consider all properties in classes that doesn't have creator function (the function that creates the object instance) as custom properties. Custom properties do not exists in the code. We create it on the fly on adding a new object instance. Here is [the code](https://github.com/surveyjs/survey-library/blob/master/src/jsonobject.ts#L524) that does the magic.
However, some our abstract classes do not have creator function as well. As result, there is no way to distingish when we need to create the custom property and when we should not.
`Serializer.addProperty`/`Serializer.addProperties` adds custom properties. They assume that there is a class (define in the code or created on fly) and you need a new custom property/properties for it.

Thank you,
Andrew

**Comment by andrewtelnov** [comment]
@SamMousa @RomanTsukanov I have implemented this functionality in code. Here is the related [PR](https://github.com/surveyjs/survey-library/pull/4732).
Json class has isCustom read-only property now. All properties are custom for isCustom class and on creating an object we add these properties on the fly. isCustom property is true if there is no creator function for the Json class. However, if there is a class that inherited from this class and a new class has a creator function, then the base class stops to be a "custom" and we update properties information accordingly.
In short, the intial code will work from v1.9.46.

Thank you,
Andrew
